core/src/test/java/com/microsoft/applicationinsights/internal/heartbeat/HeartBeatProviderMock.java

```diff
@@ -0,0 +1,114 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class HeartBeatProviderMock implements HeartBeatProviderInterface {
```

Comment:
We can do this later on too. I would like to get this in and since this is a test class and also doesn't affect performance I don't believe it is a blocker.